timely_beliefs/tests/test_belief_query.py

```diff
@@ -281,15 +303,36 @@ def _test_empty_frame(time_slot_sensor):
 def test_select_most_recent_deterministic_beliefs(
```

Comment:
It seems we're doing two independent tests here, one for most recent beliefs, one for most recent events. Can we also test using both `most_recent_only` and `most_recent_events_only` in the same within-query?

Reply:
I added your requested test, plus a docstring.